### PR TITLE
Lint missing docs in icu_datetime

### PR DESCRIPTION
**Pull request by gregtatum**
This is an initial stab at #686 for icu_datetime. There are still lots of docs to write here.

### CODE REVIEW

**Comment by sffc** [comment]
I think we should merge this with the docs Greg wrote, even if they aren't comprehensive.  It's a step in the right direction, and better to be merged than to sit in a PR.

**Comment by gregtatum** [comment]
I finished either filling in the docs or adding suppressions.

File: components/datetime/src/date.rs [hunk 1]
```diff
@@ -270,6 +288,7 @@ pub struct DayOfYearInfo {
 /// assert_eq!(7, IsoWeekday::Sunday as usize);
 /// ```
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[allow(missing_docs)] // The weekday variants should be self-obvious.
```
Comment (nordzilla) [inline components/datetime/src/date.rs:291]: >  Question

This has a doc comment. Why is the linter saying it's missing?

File: components/datetime/src/date.rs [hunk 2]
```diff
@@ -424,6 +462,8 @@ pub enum FractionalSecond {
 pub struct GmtOffset(i32);
 
 impl GmtOffset {
+    /// Attempt to create a `GmtOffset` from a seconds input. It returns an error when the seconds
```
Comment (nordzilla) [inline components/datetime/src/date.rs:465]: > Nitpick

Link ``[`GmtOffset`]``

File: components/datetime/src/provider/mod.rs [hunk 3]
```diff
@@ -6,24 +6,53 @@
...
+
+/// Data providers for time zones.
 pub mod time_zones;
 
+/// A collection of [`ResourceKey`](icu_provider::prelude::ResourceKey) structs for DateTime providers.
 pub mod key {
     use icu_provider::{resource_key, ResourceKey};
+
+    /// A [`ResourceKey`](icu_provider::prelude::ResourceKey) to
+    /// [`gregory::DatesV1`](crate::provider::gregory::DatePatternsV1).
```
Comment (nordzilla) [inline components/datetime/src/provider/mod.rs:22]: > Thought

I've been thinking a bit about our documentation style. Manually linking these keywords can be very verbose, and a pain. 

> Question

How do you feel about adding doc-only `use` declaration to the top of the file?

e.g.
```rust
#[cfg(doc)]
use {
    crate::provider::{
        gregory::DateSymbolsV1,
        time_zones::{
            ExemplarCitiesV1, MetaZoneGenericNamesLongV1, MetaZoneGenericNamesShortV1,
            MetaZoneSpecificNamesLongV1, MetaZoneSpecificNamesShortV1, TimeZoneFormatsV1,                           
        },  
    },  
    icu_provider::prelude::ResourceKey,
};
```

I think this style would really unclutter our documentation. And it's nice to have all the doc-only dependencies in one spot. I think it's also nice that it gives readers of the file some intuition of related modules, even though they aren't directly included as dependencies when building the code itself.

**Review by nordzilla — DISMISSED** [review]
LGTM, pending a few minor comments.

**Review by zbraniecki — DISMISSED** [review]

**Review by dminor — APPROVED** [review]
lgtm!

**Pull request merged**